server/db/db.js

```diff
@@ -1,826 +1,883 @@
+  const transactionMode = typeof client !== 'undefined'
+
+  try {
+    if (!transactionMode) {
+      client = await pool.connect()
+    }
+
+    const values = [installationId, buttonId, unit, phoneNumber, ALERT_STATE.STARTED, numPresses, buttonBatteryLevel]
+    const { rows } = await client.query(
+      'INSERT INTO sessions (installation_id, button_id, unit, phone_number, state, num_presses, button_battery_level) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING *',
+      values,
+    )
+
+    if (rows.length > 0) {
+      return createSessionFromRow(rows[0])
     }
-    catch(e) {
-        helpers.log(`Error running the getAllSessions query: ${e}`);
-    } finally {
-        if (!transactionMode) {
-            try {
-                client.release()
-            } catch(err) {
-                helpers.log(`getAllSessions: Error releasing client: ${err}`)
-            }
-        }
+  } catch (e) {
+    helpers.log(`Error running the createSession query: ${e}`)
+  } finally {
+    if (!transactionMode) {
+      try {
+        client.release()
+      } catch (err) {
+        helpers.log(`createSession: Error releasing client: ${err}`)
+      }
     }
-    
-    return null
+  }
+
+  return null
 }
 
-module.exports.createSession = async function(installationId, buttonId, unit, phoneNumber, numPresses, buttonBatteryLevel, client) {
-    const transactionMode = (typeof client !== 'undefined')
+async function saveSession(session, clientParam) {
+  let client = clientParam
+  const transactionMode = typeof client !== 'undefined'
 
-    try {
-        if(!transactionMode) {
-            client = await pool.connect()
-        }
-        
-        const values = [installationId, buttonId, unit, phoneNumber, ALERT_STATE.STARTED, numPresses, buttonBatteryLevel]
-        const { rows } = await client.query('INSERT INTO sessions (installation_id, button_id, unit, phone_number, state, num_presses, button_battery_level) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING *', values)
+  try {
+    if (!transactionMode) {
+      client = await pool.connect()
+    }
 
-        if(rows.length > 0) {
-            return createSessionFromRow(rows[0])
-        }
+    const { rows } = await client.query('SELECT * FROM sessions WHERE id = $1 LIMIT 1', [session.id])
+    if (rows.length === 0) {
+      throw new Error("Tried to save a session that doesn't exist yet. Use createSession() instead.")
     }
-    catch(e) {
-        helpers.log(`Error running the createSession query: ${e}`);
-    } finally {
-        if (!transactionMode) {
-            try {
-                client.release()
-            } catch(err) {
-                helpers.log(`createSession: Error releasing client: ${err}`)
-            }
-        }
+    const query =
+      'UPDATE sessions SET installation_id = $1, button_id = $2, unit = $3, phone_number = $4, state = $5, num_presses = $6, incident_type = $7, notes = $8, fallback_alert_twilio_status =$9, button_battery_level=$10 WHERE id = $11'
```

Comment:
This is weird, consensus in our code review meeting is to put the query string directly into the `client.query()` call.